pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/SortOperator.java

```diff
@@ -70,8 +72,13 @@ public SortOperator(MultiStageOperator upstreamOperator, List<RexExpression> col
     _upstreamErrorBlock = null;
     _isSortedBlockConstructed = false;
     _numRowsToKeep = _fetch > 0 ? _fetch + _offset : defaultHolderCapacity;
-    _rows = new PriorityQueue<>(_numRowsToKeep,
-        new SortComparator(collationKeys, collationDirections, dataSchema, false));
+    if (collationKeys.isEmpty()) {
+      _priorityQueue = null;
+    } else {
+      _priorityQueue = new PriorityQueue<>(_numRowsToKeep,
+          new SortComparator(collationKeys, collationDirections, dataSchema, false));
+    }
+    _rows = new ArrayList<>();
```

Comment:
I don't think this initialization is needed for both. Only for `non ORDER BY` scenario imo ?